lib/stack_master/template_compiler.rb

```diff
@@ -1,8 +1,11 @@
 module StackMaster
   class TemplateCompiler
+    class TemplateCompilationFailed < RuntimeError; end
```

Comment:
Style nitpick. We usually write this like so: `TemplateCompilationFailed = Class.new(RuntimeError)`